### PR TITLE
JAMES-3458 Reduce Cassandra query count for IMAP GetQuotaRoot queries

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Comment by chibenwa** [comment]
I HATE mocks!

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

File: mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java [hunk 1]
```diff
@@ -96,7 +96,11 @@ public boolean hasRight(Mailbox mailbox, Right right, MailboxSession session) {
     @Override
     public Rfc4314Rights myRights(MailboxPath mailboxPath, MailboxSession session) throws MailboxException {
         if (mailboxPath.belongsTo(session)) {
-            return MailboxACL.FULL_RIGHTS;
+            if (mailboxSessionMapperFactory.getMailboxMapper(session).pathExists(mailboxPath).block()) {
```
Comment (rouazana) [inline mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java:99]: are you adding one request to remove one?

Comment (chibenwa) [reply mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java:99]: No. Insteadof 2 request (mbx+acl) we only do one (mbx). This is a net win.

**Pull request closed without merging**